build-system/tasks/create-module-compatible-scripts/index.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const $$ = require('gulp-load-plugins')();
+const colors = require('ansi-colors');
+const gulp = $$.help(require('gulp'));
+const log = require('fancy-log');
+const MagicString = require('magic-string');
+const parser = require('@babel/parser');
+const through = require('through2');
+const traverse = require('@babel/traverse').default;
+
+
+function transform(file, encoding, callback) {
+  const code = file.contents.toString('utf8');
+  const ast = parser.parse(code);
+  const magicString = new MagicString(code);
+
+  traverse(ast, {
+    enter(path) {
+      if (path.node.type !== 'ConditionalExpression')
+      {return;}
+      const {node} = path;
+      const {consequent, alternate} = node;
+      if (consequent.type !== 'MemberExpression') {return;}
+      if (alternate.type !== 'ThisExpression') {return;}
+      const {object, property} = consequent;
+      if (object.name !== 'window' && property.name !== 'global') {return;}
+      magicString.overwrite(alternate.start, alternate.end, 'self');
```

Comment:
A nice pattern for these kind of changes is to collect the changes you'd like to apply, then after you've completed analysis and think the changes are safe... apply them all at once.

This lets you easily back out of applying changes if you determine something later then an earlier mutation would have had knowledge to.

Alternatively, you can also just return the original code when you bail out and keep this approach.

Reply:
done